src/others/components/CollapsibleList.tsx

```diff
@@ -12,21 +14,43 @@ export interface CollapsibleTableProps extends TableContainerProps {
 }
 
 export const CollapsibleTable = ({ rows, sortRight, renderRowData, ...tableProps }: CollapsibleTableProps) => {
-  // TODO: implement proper infinite scroll
-  const visibleRows = rows.slice(0, 30);
+  const offset = 20;
+  const getGirstBatch = (rows: ListItem[]) => rows.slice(0, offset);
+
+  const [displayedRows, setDisplayedRows] = useState<ListItem[]>([]);
+
+  const hasDisplayedAll = displayedRows.length === rows.length;
+
+  const addMoreRows = () => {
+    const nextRows = rows.slice(displayedRows.length, displayedRows.length + offset);
+    setDisplayedRows([...displayedRows, ...nextRows]);
+  };
+
+  useEffect(() => {
+    setDisplayedRows(getGirstBatch(rows));
+  }, [rows]);
+
   // TODO: type it properly - strings can be sorted too
-  const sorted = sortRight ? visibleRows.sort((a, b) => sortRight(Number(a.value), Number(b.value))) : visibleRows;
+  const sorted = sortRight ? displayedRows.sort((a, b) => sortRight(Number(a.value), Number(b.value))) : displayedRows;
```

Comment:
do i understand correctly that it will be doing partial sorts?

why not sort the data once at the very beginning and not deal with sorting again?

Reply:
No, it's a callback passed as reference, so it's doesn't matter.

Reply:
Changed, thanks!